app/models.py

```diff
@@ -2988,5 +2996,32 @@ def extend_subscription(
             },
         )
 
+    @classmethod
+    def downloaded_provider_complaint(cls, admin_user_id: int, complaint_id: int):
+        cls.create(
+            admin_user_id=admin_user_id,
+            action=AuditLogActionEnum.download_provider_complaint.value,
+            model="ProviderComplaint",
+            model_id=complaint_id,
+            data={},
+        )
+
+
+class ProviderComplaintState(EnumE):
+    new = 0
+    reviewed = 1
 
-# endregion
+
+class ProviderComplaint(Base, ModelMixin):
+    __tablename__ = "provider_complaint"
+
+    user_id = sa.Column(sa.ForeignKey("users.id"), nullable=False)
+    state = sa.Column(sa.Integer, nullable=False)
+    phase = sa.Column(sa.Integer, nullable=False)
```

Comment:
I wonder if we can set phase as an enum, something like the following?

```
 phase = sa.Column(
        sa.Enum(Phase),
        nullable=False,
        server_default=Phase.unknown.value,
    )
```

Reply:
:+1: